src/js/components/ChooseProjectModal.tsx

```diff
@@ -4,72 +4,97 @@ import { useAppSelector, useTranslationCustom, useTranslationDefault } from '@/h
 import { Link, useLocation } from 'react-router-dom';
 import { FaDatabase } from 'react-icons/fa';
 import { getCurrentPage } from '@/utils/router';
+import type { Project } from '@/types/metadata';
+
+type ChooseDatasetProps = {
+  parentProject: Project;
+  baseURL: string;
+  isSingleProject?: boolean;
+};
+const ChooseDataset = ({ parentProject, baseURL, isSingleProject }: ChooseDatasetProps) => {
```

Comment:
move this into its own file I think, and move that + the choose project modal into a folder called `scope` or `scopeChooser`